### ISSUE TITLE
no-raw-colors does not read Tailwind v4 namespaced color theme keys

### ISSUE THREAD

**Issue by simplyjackfoster** [shadcn-ui/lint#7]
### What happens

`no-raw-colors` builds its set of declared tokens from `@theme` declarations whose name starts with `color-` (`applyTokenDeclarations` in `dist/index.js`). Tailwind v4 also accepts colors under the scoped namespaces `--background-color-*`, `--text-color-*`, and `--border-color-*`, and generates `bg-<name>` / `text-<name>` / `border-<name>` from them.

In a theme that declares its semantic colors that way, every semantic utility is reported as "not a declared theme color", and the suggestion list is the raw primitive palette the theme meant to hide behind those semantics.

### Minimal reproduction

```css
@import "tailwindcss";

@theme static {
  --color-*: initial;
  --color-gray-500: #6b7280;
}

:root {
  --surface: var(--color-gray-500);
}

@theme inline {
  --background-color-surface: var(--surface);
}
```

```tsx
<div className="bg-surface" />
// reported: "bg-surface" is not a declared theme color. Use one of: gray-500 ...

<div className="bg-gray-500" />
// passes (declared under --color-*)
```

Tailwind itself generates `bg-surface`: `__unstable__loadDesignSystem(css).candidatesToCss(['bg-surface'])` returns `.bg-surface { background-color: var(--surface); }`.

### Related: `no-unknown-classes` defers color classes

`no-unknown-classes` skips an unknown class in the color category when the Tailwind worker returns no spelling suggestion, leaving the verdict to `no-raw-colors`. A project that has to turn `no-raw-colors` off for the reason above therefore gets no report at all for an invented color token such as `bg-surfaec-2`, even though the worker knows Tailwind generates nothing for it.

### Expected

- `no-raw-colors` treats names declared under the namespaced color keys as declared tokens (`--background-color-x` → `bg-x`, `--text-color-x` → `text-x`, `--border-color-x` → `border-x`, and the other namespaces Tailwind maps).
- Or `no-unknown-classes` reports an unknown color class itself when `no-raw-colors` is not enabled.

### Versions

`@shadcn/lint` 0.1.0, `tailwindcss` 4.3.x, `oxlint` 1.81 (`jsPlugins`).

**Comment by jkinley** [comment]
I tested and this seems to be true. The linter should only allow semantic token colors. muted, secondary, primary, etc.
